test/tests/query_params_test.js

```diff
@@ -3,8 +3,24 @@ import Router from "router";
 import { resolve, configure, reject, Promise } from "rsvp";
 
 var router, url, handlers, expectedUrl, actions;
+var scenarios = [
+  {
+    name: 'Sync Get Handler',
+    getHandler: function(name) {
+      return handlers[name] || (handlers[name] = {});
+    }
+  },
+  {
+    name: 'Async Get Handler',
+    getHandler: function(name) {
+      return Promise.resolve(handlers[name] || (handlers[name] = {}));
```

Comment:
So turns out the tests setup a custom async handler for RSVP ([see here](https://github.com/tildeio/router.js/blob/master/test/tests/test_helpers.js#L10)), which seems to cause problems when used in conjunction with `setTimeout` since Backburner gets flushed independent of when the timeouts resolve.

I tend to think that we should proceed as is.